.github/workflows/release.yaml

```diff
@@ -1,186 +1,192 @@
+    #       path: dist
+    #   - uses: actions/download-artifact@v1
+    #     with:
+    #       name: server-macos-latest
+    #       path: dist
+    #   - uses: actions/download-artifact@v1
+    #     with:
+    #       name: server-ubuntu-latest
+    #       path: dist
+    #   - uses: actions/download-artifact@v1
+    #     with:
+    #       name: server-windows-latest
+    #       path: dist
+    #   - run: ls -all ./dist
 
       - run: echo "::set-env name=TAG::$(date --iso)"
       - run: 'echo "TAG: $TAG"'
 
-      - name: Create Release
-        id: create_release
-        # uses: actions/create-release@v1
-        # https://github.com/actions/create-release/pull/32
-        uses: fleskesvor/create-release@1a72e235c178bf2ae6c51a8ae36febc24568c5fe
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ env.TAG }}
-          release_name: ${{ env.TAG }}
-          draft: false
-          prerelease: false
-
-      - uses: actions/upload-release-asset@v1.0.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./dist/ra_lsp_server-linux
-          asset_name: ra_lsp_server-linux
-          asset_content_type: application/octet-stream
-
-      - uses: actions/upload-release-asset@v1.0.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./dist/ra_lsp_server-mac
-          asset_name: ra_lsp_server-mac
-          asset_content_type: application/octet-stream
-
-      - uses: actions/upload-release-asset@v1.0.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./dist/ra_lsp_server-windows.exe
-          asset_name: ra_lsp_server-windows.exe
-          asset_content_type: application/octet-stream
-
-      - uses: actions/upload-release-asset@v1.0.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./dist/rust-analyzer-0.1.0.vsix
-          asset_name: rust-analyzer-0.1.0.vsix
-          asset_content_type: application/octet-stream
-
-      - uses: actions/upload-release-asset@v1.0.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./dist/rust-analyzer.el
-          asset_name: rust-analyzer.el
-          asset_content_type: text/plain
+      - name: Publish Extension
+        working-directory: ./editors/code
+        # token from https://dev.azure.com/rust-analyzer/
+        run: ./node_modules/vsce/out/vsce publish --pat ${{ secrets.MARKETPLACE_TOKEN }} 0.1.0-$TAG
```

Comment:
Maybe use 0.1-$TAG instead of 0.1.0-$TAG as version?

Reply:
The sevmer needs three digits I think.

Reply:
Using a date is not semver anyway, and it doesn't seem right to keep the patch level equal for every single release, even when adding new features or removing them.

Reply:
It is sevmver: you can attach arbitrary tags after `-`:

https://semver.org/#spec-item-9

Reply:
I mean that normal semver compatibility is not followed by using it like this with a date as only changing thing, instead of categorizing releases into major, minor and patch releases.

Reply:
That's true. I mean that it is syntactically semver, and I think package.json requires this field to syntactically be a semver.

Reply:
Who needs standards anyway:

>  ERROR  The version string '0.1.0-2020-02-11' doesn't conform to the requirements for a version. It must be one to four numbers in the range 0 to 2147483647, with each number seperated by a period. It must contain at least one non-zero number.

https://github.com/rust-analyzer/rust-analyzer/runs/438183308?check_suite_focus=true#step:7:22

:D

Reply:
Bhahaha:

> ERROR  Invalid version 0.1.0.20200211